sklearn/manifold/_t_sne.py

```diff
@@ -22,7 +22,8 @@
 from ..decomposition import PCA
 from ..metrics.pairwise import pairwise_distances
 from . import _utils
-from . import _barnes_hut_tsne
+# mypy error: Module 'sklearn.manifold' has no attribute '_barnes_hut_tsne'
+from . import _barnes_hut_tsne  # type: ignore
```

Comment:
Why is `_barnes_hut_tsne` causing a typing error?

Reply:
> Why is _barnes_hut_tsne causing a typing error?

There is some mypy issue with importing full C extension modules (but not individual objects from these modules). Couldn't find relevant information in the mypy issue tracker and opened https://github.com/python/mypy/issues/8575

Reply:
Summary of the above linked mypy issue is that's expected behavour,

> Mypy doesn't actually import or interact with C extension code. It can only parse and gain type information from .py and .pyi files.

so it sounds like all C extensions need to be ignored for typing in some way, either explicitly with `type: ignore` or via the `--ignore-missing-import` option.

@WillAyd since you worked on this in pandas, what's the current approach for mypy and C extensions there? As far as I can tell pandas doesn't use neither of the above options, and yet I don't see any C extension related errors with `mypy pandas` or errors about missing stabs for dependencies for that matter such as,
```
sklearn/cluster/_spectral.py:10: error: Skipping analyzing 'numpy': found module but no type hints or library stubs
```
that we currently skip in this PR using `--ignore-missing-import`. Thanks!

Reply:
For better or worse we set `ignore_missing_imports=True` in our config, so that this problem is "solved" globally rather than having to ignore individual imports

Reply:
Ahh, indeed, finally found [pandas mypy config](https://github.com/pandas-dev/pandas/blob/5b751ec79d17722423a170778d028b283ed2c47e/setup.cfg#L124). Was previously grepping option names with "-" instead of "_". Thanks for confirming @WillAyd!

For the record, the above mentioned `# type: ignore` is still necessary since this particular import doesn't get skipped by the global `--ignore-missing-import` option as discussed in the above linked mypy issue.